code/modules/mob/living/simple_animal/hostile/hivebot.dm

```diff
@@ -31,85 +68,207 @@
 	max_n2 = 0
 	minbodytemp = 0
 	speed = 4
-
 	tameable = FALSE
+	flying = 1
+	see_in_dark = 8
+	pass_flags = PASSTABLE
+	attack_emote = "focuses on"
+	var/mob/living/simple_animal/hostile/hivebotbeacon/linked_parent = null
 
 /mob/living/simple_animal/hostile/hivebot/range
 	name = "Hivebot"
-	desc = "A smallish robot, this one is armed!"
+	desc = "A primitive in design, hovering robot, with a simple looking launcher sticking out of it. It bears no manufacturer markings of any kind."
+	icon_state = "hivebotranged"
 	ranged = 1
 	smart = TRUE
 
 /mob/living/simple_animal/hostile/hivebot/range/rapid
+	projectiletype = /obj/item/projectile/bullet/pistol/hivebotspike/needle
 	rapid = 1
 
-/mob/living/simple_animal/hostile/hivebot/range/strong
-	name = "Strong Hivebot"
-	desc = "A robot, this one is armed and looks tough!"
-	health = 80
+//Creates a reference to its parent beacon on init.
+/mob/living/simple_animal/hostile/hivebot/Initialize(mapload,mob/living/simple_animal/hostile/hivebot/hivebotbeacon)
+	if(hivebotbeacon)
+		linked_parent = hivebotbeacon
+		linked_parent.existing_bots += 1
+	.=..()
 
 /mob/living/simple_animal/hostile/hivebot/death()
-	..()
-	visible_message("<b>[src]</b> blows apart!")
-	new /obj/effect/decal/cleanable/blood/gibs/robot(src.loc)
-	spark(src, 3, alldirs)
+	..(null,"blows apart!")
+	if(linked_parent)

```

Comment:
Put this code (lines 98-100) into `Destroy()` instead. It's less likely to not be called from there.